lib/compat/wordpress-6.7/blocks.php

```diff
@@ -103,3 +103,31 @@ static function ( $format ) {
 	return $query;
 }
 add_filter( 'query_loop_block_query_vars', 'gutenberg_add_format_query_vars_to_query_loop_block', 10, 2 );
+
+/**
+ * Adds `ignore` option for sticky posts to the Query block.
+ * 
+ * @see 'query_loop_block_query_vars'
+ * 
+ * @param array    $query The query vars.
+ * @param WP_Block $block Block instance.
+ * @return array   The filtered query vars.
+ */
+function gutenberg_add_ignore_sticky_posts_to_query_loop_block( $query, $block ) {
+	if ( ! empty( $block->context['query']['sticky'] ) && 'ignore' === $block->context['query']['sticky'] ) {
+		// Core function excludes all sticky posts if the `sticky` value is anything
+		// other than `only`. We must reset that here, but it could potentially also
+		// re-allow a sticky post that had been excluded in some other way. This 
+		// works okay for testing, but the real fix will need to be in the
+		// core function.
+		$sticky = get_option( 'sticky_posts' );
+
+		$query['post__not_in'] = array_diff( $query['post__not_in'], ! empty( $sticky ) ? $sticky : array() );
```

Comment:
Why do we exlude sticky posts? Ignore set below should include them and just list them in their natural order, right?

Reply:
Line 125 doesn't exclude them, it's actually re-including them after core already excluded them. That's what the comment above line 123 is about.